stubs/urllib3/urllib3/util/retry.pyi

```diff
@@ -12,16 +12,25 @@ ResponseError = exceptions.ResponseError
 log: Any
 
 class Retry:
-    DEFAULT_METHOD_WHITELIST: Any
-    BACKOFF_MAX: Any
-    total: Any
-    connect: Any
-    read: Any
-    redirect: Any
-    status_forcelist: Any
-    method_whitelist: Any
-    backoff_factor: Any
-    raise_on_redirect: Any
+    DEFAULT_ALLOWED_METHODS: ClassVar[frozenset[str]]
+    RETRY_AFTER_STATUS_CODES: ClassVar[frozenset[int]]
+    DEFAULT_REMOVE_HEADERS_ON_REDIRECT: ClassVar[frozenset[str]]
+    DEFAULT_BACKOFF_MAX: ClassVar[int]
+
+    total: bool | int | None
+    connect: int | None
+    read: int | None
+    redirect: bool | int | None
+    status: int | None
+    other: int | None
+    allowed_methods: Collection[str] | None
+    status_forcelist: Collection[int] | None
+    backoff_factor: float
+    raise_on_redirect: bool
+    raise_on_status: bool
+    history: tuple[Any] | None
+    respect_retry_after_header: bool
+    remove_headers_on_redirect: Collection[str]
```

Comment:
Any `Collection` can be passed into `__init__`, but it is [changed to be a `frozenset`](https://github.com/urllib3/urllib3/blob/33afb5c96c435ab7118b5f41e26f96c290fef55b/src/urllib3/util/retry.py#L239) when it is set as an attribute on the class.

```suggestion
    remove_headers_on_redirect: frozenset[str]
```